common/src/main/java/com/wynntils/core/webapi/WebManager.java

```diff
@@ -231,12 +237,89 @@ public static boolean tryLoadItemList() {
                 .useCacheAsBackup()
                 .build());
 
-        handler.dispatch(false);
-
         // Check for success
         return isItemListLoaded();
     }
 
+    public static CompletableFuture<Boolean> tryLoadMap() {
+        if (apiUrls == null || !apiUrls.hasKey("MainMap")) return CompletableFuture.completedFuture(false);
+
+        final File mapDirectory = new File(API_CACHE_ROOT, "map");
+
+        mapDirectory.mkdirs();
+
+        final CompletableFuture<Boolean> result = new CompletableFuture<>();
+
+        handler.addAndDispatchAsync(new RequestBuilder(apiUrls.get("MainMap"), "main_map.info")
```

Comment:
I am very, very skeptical towards bringing this old bad `CenterX` thing into Artemis.

I want to have a new, json-based metadata file with map information. I propose this:
`maps.json`:
```
[
{ "file": "main-map.png",
  "x1": -3443,
  "z1": 1002,
  "x2": 2443,
  "z2": -3922
} 
]
```

This is interpreted that one tile of the map (in this case, the entire world map) resides in the file `main-map.png`. The top left pixel on the map png corresponds to the block at the position X=-3443,Z=1002, and the bottom right corresponds to X=2443,Z=-3922. It is a failure if the size of the png does not match the size spanned by the coordinates 2D BB, as given in the metadata.

Reply:
@magicus would we want to limit ourselves to 1 pixel = 1 block? We could have an option for higher resolution maps in the future. Also, should we add hashes to the metadata? That way we only have to download new/updated areas. Although that might prevent users from using a custom map.

Reply:
I agree, but this is not the pr to change things. `CenterX` and `CenterZ` actually refer to the bottom right X and Z coordinates so they have been renamed here.

Reply:
You mean that it would require changes to Athena? 

I still think this *is* the place to introduce such changes. It's better that you make a proper json file, even if you have to check it in as part of the resources of Artemis, than to write code that parses to old and bad stuff.

I also think that we make the Athena/server part faaaar to complicated. For static data caching, we just need to fetch it off a github repo. Wham, bam, done. The Athena logic is really only needed for stuff like user tracking, etc.

So better yet, make a json file and put it at [Data-Storage](https://github.com/Wynntils/Data-Storage) in e.g. a new `Artemis` directory, and get it using http from a raw link like `https://raw.githubusercontent.com/Wynntils/Data-Storage/master/item-changes.json`.

Reply:
And yes, @HighCrit's idea of a hash is an excellent idea. If we download the metadata, and the hash matches the png file we have on disk in our cache, we're good, otherwise we need to refetch it.

Reply:
@HighCrit Acutally, my first attempt at the json format included a `"scale": 1`, but then I removed it. I think we better start with that assumption. If we ever want to change it, it is easy to add the scale factor back. But in the meantime, we will be creating a lot of extra work for ourselves, trying to pass around a scale factor that is not used. And in the end, if it is not tested, it will likely be broken anyway the day we introduce a non-1-to-1 map. So my recommendation is: "Yes, it might be a good idea, but not just yet".

Reply:
The current implementation of map fetching on the serverside resides in [WynntilsWebsite-API](https://github.com/Wynntils/WynntilsWebsite-API/tree/master/maps).

Reply:
This should be added now, with the additional pr of https://github.com/Wynntils/WynntilsWebsite-API/pull/7.